lib/pmdtester/builders/summary_report_builder.rb

```diff
@@ -0,0 +1,146 @@
+      details
+    end
+
+    def build_body(doc)
+      build_branch_details_section(doc)
+      build_projects_section(doc)
+    end
+
+    def build_branch_details_section(doc)
+      doc.div(class: 'section', name: 'Branch details') do
+        doc.h2 'Branch details:'
+        build_branch_details_table(doc)
+      end
+    end
+
+    def build_branch_details_table(doc)
+      doc.table(class: 'bodyTable', border: '0') do
+        build_branch_table_head(doc)
+        build_branch_table_body(doc)
+      end
+    end
+
+    def build_branch_table_head(doc)
+      doc.thead do
+        doc.tr do
+          doc.th 'Item'
+          doc.th 'base'
+          doc.th 'patch'
+        end
+      end
+    end
+
+    def build_branch_table_body(doc)
+      doc.tbody do
+        build_branch_table_row(doc, 'branch name', @base_details.branch_name,
+                               @patch_details.branch_name)
+        build_branch_table_row(doc, 'branch last commit sha', @base_details.branch_last_sha,
+                               @patch_details.branch_last_message)
+        build_branch_table_row(doc, 'branch last commit message', @base_details.branch_last_message,
+                               @patch_details.branch_last_message)
+        build_branch_table_row(doc, 'total execution time', @base_details.execution_time,
+                               @patch_details.execution_time)
+        build_branch_config_table_row(doc)
+      end
+    end
+
+    def build_branch_config_table_row(doc)
+      doc.tr do
+        doc.td(class: 'c') { doc.text 'branch configuration' }
+        base_config_src_path = @base_details.target_branch_config_path
+        copy_branch_config_file(base_config_src_path, BASE_CONFIG_PATH)
+        doc.td(class: 'a') do
+          doc.a(href: './base_config.xml') { doc.text 'base config' }
+        end
+        patch_config_stc_path = @patch_details.target_branch_config_path
+        FileUtils.cp(patch_config_stc_path, PATCH_CONFIG_PATH)
+        doc.td(class: 'b') do
+          doc.a(href: './patch_config.xml') { doc.text 'patch config' }
+        end
+      end
+    end
+
+    def copy_branch_config_file(src, dest)
+      FileUtils.cp(src, dest) if File.exist?(src)
+    end
+
+    def build_branch_table_row(doc, item, base, patch)
+      doc.tr do
+        doc.td(class: 'c') { doc.text item }
+        doc.td(class: 'a') { doc.text base }
+        doc.td(class: 'b') { doc.text patch }
+      end
+    end
+
+    def build_projects_section(doc)
+      doc.div(class: 'section', name: 'Projects') do
+        doc.h2 'Projects:'
+        build_projects_table(doc)
+      end
+    end
+
+    def build_projects_table(doc)
+      doc.table(class: 'bodyTable', border: '0') do
+        build_projects_table_head(doc)
+        build_projects_table_body(doc)
+      end
+    end
+
+    def build_projects_table_head(doc)
+      doc.thead do
+        doc.tr do
+          doc.th 'project diff report link'
+          doc.th 'project name'
```

Comment:
I would actually remove the first column and make the project name the link to the reports.

Reply:
Done.